src/Symfony/Component/Security/Core/User/InMemoryUser.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\User;
+
+/**
+ * UserInterface implementation used by the in-memory user provider.
+ *
+ * This should not be used for anything else.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class InMemoryUser implements UserInterface, PasswordAuthenticatedUserInterface, EquatableInterface
+{
+    private $username;
+    private $password;
+    private $enabled;
+    private $roles;
+
+    /**
+     * @param string[] $roles
+     */
+    public function __construct(?string $username, ?string $password, array $roles = [], bool $enabled = true)
```

Comment:
```suggestion
    public function __construct(string $username, ?string $password, array $roles = [], bool $enabled = true)
```
Or is that not possible for BC ?

Reply:
It is possible :) I need to double check that there is no legit use case for `null`, but 👍 . Thanks @rosier!